crates/circuit/src/dag_circuit.rs

```diff
@@ -2216,6 +2233,182 @@ impl DAGCircuit {
         })
     }
 
+    /// Are these two DAGs structurally equal?
+    ///
+    /// This function returns true iff the graph structures are precisely the same as each other,
+    /// including the valid node indices, edge orders, and so on.  This is a much stricter check
+    /// than graph equivalence, and is mostly useful for testing if two :class:`DAGCircuit`
+    /// instances have been constructed and manipulated in the exact same ways.  For example, this
+    /// method can be used to test whether a sequence of manipulations of a DAG is deterministic.
+    ///
+    /// This method does not consider tracking metadata such as :attr:`metadata` or :attr:`name`,
+    /// but does consider many low-level implementation details of the internal representation, many
+    /// of which do not change the semantics of the circuit.
+    ///
+    /// ..
+    ///
+    ///     Note to devs: the internal details referred to are things like the interners.
+    ///
+    /// This method should, in general, be much faster than graph-equivalence checks, but will
+    /// return ``False`` in many more situations.  This method should never return ``True`` when a
+    /// graph-equivalence check would return ``False``.
```

Comment:
I might put this comment at the bottom of the admonition, although realistically I would probably just make this a literal comment instead of a sphix comment with `//` but it doesn't matter that much.

Reply:
I actually just deleted the Sphinx comment - I don't know what I was thinking, because this doesn't compare interner keys directly.

Reply:
Done in e28c8d225a.